napari/utils/events/containers/_selection.py

```diff
@@ -25,47 +32,99 @@ class Selection(EventedSet[_T]):
-        Will not be emitted if the item was not in the set when discarded.
-    current (value: _T, previous: _T)
+    changed (added: Set[_T], removed: Set[_T])
+        Emitted when the set changes, includes item(s) that have been added
+        and/or removed from the set.
+    current (value: _T)
+        emitted when the current item has changed.
+    active (value: _T)
         emitted when the current item has changed.
     """
 
-    def __init__(self, data: Iterable[_T] = (), current: Optional[_T] = None):
+    def __init__(self, data: Iterable[_T] = ()):
+        self._active: Optional[_T] = None
+        self.__current = None
         super().__init__(data=data)
-        self.events.add(current=None)
-        self._current = current
+        self.events.add(_current=None, active=None)
+        self.events.changed.connect(self._update_active)
+        self._update_active()
 
     def __repr__(self) -> str:
-        clsname = type(self).__name__
-        return f"{clsname}({repr(self._set)}, current={self.current})"
+        return f"{type(self).__name__}({repr(self._set)})"
+
+    def __hash__(self) -> int:
+        """Make selection hashable."""
+        return id(self)
 
     @property
-    def current(self) -> Optional[_T]:
+    def _current(self) -> Optional[_T]:
         """Get current item."""
-        return self._current
+        return self.__current
 
-    @current.setter
-    def current(self, index: Optional[_T]):
+    @_current.setter
+    def _current(self, index: Optional[_T]):
         """Set current item."""
-        if index == self._current:
+        if index == self.__current:
+            return
+        self.__current = index
+        self.events._current(value=index)
+
+    @property
+    def active(self) -> Optional[_T]:
+        """Return the currently active item or None."""
+        return self._active
+
+    @active.setter
+    def active(self, value: Optional[_T]):
+        """Set the active item.
+
+        This make `value` the only selected item, and make it current.
+        """
+        if value == self._active:
             return
-        previous, self._current = self._current, index
-        self.events.current(value=index, previous=previous)
+        self._active = value
+        self.clear() if value is None else self.select_only(value)
+        self._current = value
+        self.events.active(value=value)
+
+    def _update_active(self, event=None):
+        """On a selection event, update the active item based on selection.
+
+        (An active item is a single selected item).
+        """
+        if len(self) == 1:
+            self.active = list(self)[0]
+        else:
+            if self._active is not None:
+                self._active = None
+                self.events.active(value=None)
+
+    def clear(self, keep__current: bool = False) -> None:
+        """Clear the selection."""
+        if not keep__current:
+            self._current = None
+        super().clear()
```

Comment:
```suggestion
    def clear(self, keep_current: bool = False) -> None:
        """Clear the selection."""
        if not keep_current:
            self._current = None
        super().clear()
```

Probably an autoreplace snafu? ;)